connect.go

```diff
@@ -131,20 +139,28 @@ func Connect(connectStr string) (success bool) {
 		waitBusy(adTNC)
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
+	dialCancelFunc = func() { dialing = nil; cancel() }
```

Comment:
Are you sure you want this on clean disconnect, too?

Reply:
Are you referring to the defer dialCancelFunc()? It's a noop to cancel an expired context, in fact it is [required](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/context/context.go;l=230) in order to avoid leaks 🙂